Sprint-1/implement/dedupe.test.js

```diff
@@ -13,15 +13,41 @@ E.g. dedupe([1, 2, 1]) target output: [1, 2]
 
 // Acceptance Criteria:
 
-// Given an empty array
-// When passed to the dedupe function
-// Then it should return an empty array
-test.todo("given an empty array, it returns an empty array");
+describe("dedup Tests", () => {
+  // Given an empty array
+  // When passed to the dedupe function
+  // Then it should return an empty array
+  test("Given an empty array, it returns an empty array", () => {
+    const input = dedupe([]);
+    const expected = [];
+    expect(input).toEqual(expected);
+  });
 
-// Given an array with no duplicates
-// When passed to the dedupe function
-// Then it should return a copy of the original array
+  // Given an array with no duplicates
+  // When passed to the dedupe function
+  // Then it should return a copy of the original array
```

Comment:
The requirement here says returns a _copy_. But this test doesn't seem to test that a copy is returned - if you just returned the original array, I think this test would still pass.

Can you think of a way of verifying we actually returned a copy? Maybe break your implementation to return the original array, and use that to test your test, before fixing your code again.

Reply:
I've updated my testing to do a check but I'm not sure its exactly what you meant.